go-app/main.go

```diff
@@ -172,3 +176,63 @@ func loadAnalyseLog(w http.ResponseWriter, r *http.Request) {
 	}
 	reportTempl.Execute(w, report.FullLogDetails.Analysis_details)
 }
+func loadEventDetails(w http.ResponseWriter, r *http.Request) {
+	r.ParseMultipartForm(10 << 20)
+	startIndex, _ := strconv.Atoi(r.FormValue("StartIndex"))
+	endIndex, _ := strconv.Atoi(r.FormValue("EndIndex"))
+	logs := strings.Split(report.FullLogDetails.Analysis_details.RawLog, "\n")
+	type Reponse struct {
+		Content   []string
+		Highlight map[int]bool //index=> true = must be highlight
+	}
+	event_content := []string{}
+	highlight := map[int]bool{}
+	if startIndex < 0 {
+		startIndex = 0
+	}
+	if endIndex > len(logs) {
+		endIndex = len(logs)
+	}
+	event_content = fillEventDetails(startIndex, endIndex, logs, event_content, highlight)
+	resp := Reponse{
+		Content:   event_content,
+		Highlight: highlight,
+	}
+	jsonValue, _ := json.Marshal(resp)
+	w.Write(jsonValue)
+
+}
+func fillEventDetails(startIndex int, endIndex int, logs []string, event_content []string, highlight map[int]bool) []string {
+	last_index := startIndex
+	for _, index := range report.OrderedEventsLine {
```

Comment:
We're referencing the OrderedEventsLine variable in this file which is populated in another file. These kinds of globally visible variables should be discouraged. Is there any way to pass it as a parameter potentially? It's weird because we're handling all of the paths here but that includes screens which are several pages deep.